.github/workflows/test_cpu.yml

```diff
@@ -0,0 +1,79 @@
+name: Run tests on CPU
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    types: [open, reopened, synchronize, review_requested, ready_for_review]
+  schedule:
+    - cron: "0 4 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests-cpu:
+    name: ${{ matrix.os }} - python-${{ matrix.python-version }}, torch-${{ matrix.pytorch-version }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ['3.7', '3.8']
+        pytorch-version: ['lts', '1.10.2', '1.12.1', 'nightly']
+
+    steps:
+      - name: Checkout kornia
+        uses: actions/checkout@v3
+
+      - name: Setting environment on ${{ matrix.os }} with python ${{ matrix.python-version }} and pytorch  ${{ matrix.pytorch-version }}
+        uses: ./.github/actions/env
+        with:
+          python-version: ${{ matrix.python-version }}
+          pytorch-version: ${{ matrix.pytorch-version }}
+
+      - name: Run CPU tests (float32,float64)
+        shell: bash -l {0}
+        run: pytest -v --device cpu --dtype float32,float64 --cov=kornia --cov-report xml ./test
+
+      - name: Upload coverage
+        uses: codecov/codecov-action@v3
+        with:
+          file: coverage.xml
+          flags: cpu,${{ matrix.os }}_python-${{ matrix.python-version }}_torch-${{ matrix.pytorch-version }}
+          name: cpu-coverage
+
+  tests-macos:
+    name: ${{ matrix.os }} - python-${{ matrix.python-version }}, torch-${{ matrix.pytorch-version }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+        python-version: ['3.7', '3.8']
+        pytorch-version: ['1.10.2', '1.12.1', 'nightly']
+
+    steps:
+      - name: Checkout kornia
+        uses: actions/checkout@v3
+
+      - name: Setting environment on ${{ matrix.os }} with python ${{ matrix.python-version }} and pytorch  ${{ matrix.pytorch-version }}
+        uses: ./.github/actions/env
+        with:
+          python-version: ${{ matrix.python-version }}
+          pytorch-version: ${{ matrix.pytorch-version }}
+
+      - name: Run CPU tests (float32,float64)
+        shell: bash -l {0}
+        run: pytest -v --device cpu --dtype float32,float64 --cov=kornia --cov-report xml ./test
```

Comment:
Ideally we should also include half precision torch.float16

Reply:
I will add this check if it causes fails at CI -- locally i get many fails 😅

Reply:
float16 is failing, i will try another approach to add this to the CI now, but don't fail the job

Reply:
perfect